_posts/2020-12-05-data-quality-with-r.md

```diff
@@ -0,0 +1,2415 @@
+an R-script that transforms it into a table that can be used for statistical analysis.
+Since the original format is in JSON there is a risk that the format might change,
+so data validation of this table seems like a natural step.
+
+    tbl(con, in_schema("PUBLIC", "METERING_READINGS_TBL")) %>% 
+      select(-SOURCE_INSTANCE_NAME)
+
+    ## # Source:   lazy query [?? x 6]
+    ## # Database: Snowflake 4.40.1[filipw@Snowflake/xxx]
+    ##    IMPORT_TIMESTAMP    TIMESTAMP           METERING_POINT_… PROPERTY
+    ##    <dttm>              <dttm>              <chr>            <chr>   
+    ##  1 2020-10-31 02:43:48 2020-07-04 12:00:00 cf767121-4f64-4… flow_te…
+    ##  2 2020-10-31 02:43:48 2020-07-04 12:00:00 cf767121-4f64-4… return_…
+    ##  3 2020-10-31 02:43:48 2020-07-04 13:00:00 cf767121-4f64-4… flow_te…
+    ##  4 2020-10-31 02:43:48 2020-07-04 13:00:00 cf767121-4f64-4… return_…
+    ##  5 2020-10-31 02:43:48 2020-07-04 14:00:00 cf767121-4f64-4… flow_te…
+    ##  6 2020-10-31 02:43:48 2020-07-04 14:00:00 cf767121-4f64-4… return_…
+    ##  7 2020-10-31 02:43:48 2020-07-04 15:00:00 cf767121-4f64-4… flow_te…
+    ##  8 2020-10-31 02:43:48 2020-07-04 15:00:00 cf767121-4f64-4… return_…
+    ##  9 2020-10-31 02:43:48 2020-07-04 16:00:00 cf767121-4f64-4… flow_te…
+    ## 10 2020-10-31 02:43:48 2020-07-04 16:00:00 cf767121-4f64-4… return_…
+    ## # … with more rows, and 2 more variables: UNIT_OF_MEASURE <chr>, VALUE <dbl>
+
+## pointblank usage
+
+In `pointblank` we first create an `agent` that defines which table we
+want to validate. Then we use a `validation function` (these can also be
+tests and expectations) to do our validation, e.g. `col_is_character()`.
+Lastly we use the function `interrogate()` to do the actual validation.
+
+So for this table we might be interested in checking the format of the
+columns:
+
+    library(pointblank)
+    agent <- 
+      create_agent(
+          read_fn = ~tbl(con, in_schema("PUBLIC", "METERING_READINGS_TBL")),
+          tbl_name = "METERING_READINGS_TBL"
+        ) %>%
+      col_is_character(vars(METERING_POINT_ID, PROPERTY, UNIT_OF_MEASURE)) %>% 
+      col_is_numeric(vars(VALUE))%>%
+      col_is_posix(vars(IMPORT_TIMESTAMP, TIMESTAMP)) %>% 
+      interrogate()
+
+When the “interrogation” is completed we can print the agent and we’ll
+get a pretty table telling us what has happened.
+
+    agent
+
+<!--html_preserve-->
+<style>@import url("https://fonts.googleapis.com/css2?family=IBM+Plex+Sans:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap");
+@import url("https://unpkg.com/balloon-css/balloon.min.css");
+@import url("https://fonts.googleapis.com/css2?family=IBM+Plex+Mono&display=swap");
+html {
+  font-family: 'IBM Plex Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
+}
+
+#report .gt_table {
```

Comment:
This huge style part renders incorrectly in GitHub preview, like this:

![image](https://user-images.githubusercontent.com/6450386/102215289-2f2de500-3ee2-11eb-8d2a-a32247e40244.png)


If you tested in Docker and it works there it's probably okay but best to verify this part immediately when this is published, since unfortunately Docker has some differences to how end results are rendered.